pkg/aggregator/aggregator.go

```diff
@@ -171,10 +177,16 @@ type BufferedAggregator struct {
 // NewBufferedAggregator instantiates a BufferedAggregator
 func NewBufferedAggregator(s *serializer.Serializer, hostname, agentName string, flushInterval time.Duration) *BufferedAggregator {
 	aggregator := &BufferedAggregator{
-		dogstatsdIn:        make(chan *metrics.MetricSample, 100), // TODO make buffer size configurable
-		checkMetricIn:      make(chan senderMetricSample, 100),    // TODO make buffer size configurable
-		serviceCheckIn:     make(chan metrics.ServiceCheck, 100),  // TODO make buffer size configurable
-		eventIn:            make(chan metrics.Event, 100),         // TODO make buffer size configurable
+		bufferedMetricIn:       make(chan []*metrics.MetricSample, 100), // TODO make buffer size configurable
+		bufferedServiceCheckIn: make(chan []*metrics.ServiceCheck, 100), // TODO make buffer size configurable
+		bufferedEventIn:        make(chan []*metrics.Event, 100),        // TODO make buffer size configurable
+
+		metricIn:       make(chan *metrics.MetricSample, 100), // TODO make buffer size configurable
+		serviceCheckIn: make(chan metrics.ServiceCheck, 100),  // TODO make buffer size configurable
```

Comment:
To be consistent if it's possible, it would probably be better to have a chan on a pointer here and the line above?

```
		serviceCheckIn: make(chan *metrics.ServiceCheck, 100),  // TODO make buffer size configurable
		eventIn:        make(chan *metrics.Event, 100),         // TODO make buffer size configurable
```

It may even be useful as `metrics.ServiceCheck` and especially `metrics.Event` are big structs.

Reply:
Created a new PR for this https://github.com/DataDog/datadog-agent/pull/3021 (thought it would imply much more changes)

Reply:
Totally agree... luckily the volume for service checks and events is way lower, so not the end of the world to keep like this until #3021 is ready.